### ISSUE TITLE
Permalinks in documentation

### ISSUE THREAD

**Issue by olivere** [elastic/elasticsearch#20907]
As the author of an Elasticsearch client, I need a way to link to features in the various ES versions via a permalink. However, links seem to break sometimes.

E.g. I expected a link like [`https://www.elastic.co/guide/en/elasticsearch/reference/2.x/docs-bulk.html`](https://www.elastic.co/guide/en/elasticsearch/reference/2.x/docs-bulk.html) to always show the documentation for bulk operations in ES 2.x. However, the link above 404's but this one does work [`https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-bulk.html`](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-bulk.html). However, it's a PITA to change links based on the current version of ES 2.x.

Of course I expect links to break between major versions (as features change etc.), but not between minor versions.

What's the right strategy here?

**Comment by jasontedor** [comment]
The trees like 2.x are our active development branch during the lifecycle of the 2.x series. We will cut branches like 2.4 from this when preparing to stabilize the branch for a release. You'll see the same thing for the 5.x series, we have a 5.x branch that is under active development and a 5.0 branch that is stable as we push towards the 5.0.0 release. Eventually there will be a 5.1 branch cut from 5.x.

So basically, if you want relatively stable links, you should not be linking to active trees like the 2.x trees. Note that the docs are living though, we do make changes to them in the stable branches and that does from time to time mean that links to these could be broken as well.

**Comment by olivere** [comment]
Okay. I understand. 

Just for clarification. There's no way to symlink to something like `2.<current-minor-version>`? I have to change my links constantly from e.g. `2.4` to `2.5` when 2.5 arrives?

**Comment by jasontedor** [comment]
> There's no way to symlink to something like `2.<current-minor-version>`? I have to change my links constantly from e.g. `2.4` to `2.5` when 2.5 arrives?

That doesn't protect you from broken links though, the moment that 2.4 is released it's been cut from the 2.x development branch and all of your links suddenly change from pointing to things in the 2.3 tree to things in the 2.4 tree and you have no guarantees whatsoever about the stability there.

**Comment by olivere** [comment]
Yeah, fair enough. You cannot prevent broken links, and that's okay. However, it's not very likely that things change big time between 2.4 and 2.5, or is it?

I'm also not trying to push you to do anything. I just wanted to understand if I'm missing something. If you say there isn't a way, there isn't a way... :-)
